### ISSUE TITLE
I can not install robotjs on the node

### ISSUE THREAD

**Issue by BCE8E8** [Robot/node-robot#42]
when I try to install by the command: npm install robot-js, the screen shows this error

Error: robot-js binaries failed to install. You will need to install them manually.

when I put the command: npm list robotjs, this message appears:

! - UNMET DEPENDENCY robotjs@0.4.5

npm ERR! missing: robotjs@0.4.5, required by iamdinosaur@1.0.0

I do not know how to install manually and by npm is not installing

**Comment by ahsanatha** [comment]
i got same problem

**Comment by karliky** [comment]
Robot is distributed as a pre-compiled binary. We do this so that you don't have to compile the binary from the source code. When you install Robot, the binary is downloaded from `http://node.getrobot.net/addon/_version_/`. Based on the following variables.
```
var SOURCE = "http://node.getrobot.net/addon/1.0.2/" +
	process.platform   + "-" +
	process.arch       + "-" +
	process.versions.modules +
	".node";
```
Your Node.js/NPM versions should match any of the following binaries listed here:
https://github.com/Robot/robot-js-binaries/tree/gh-pages/addon/1.0.2

You can check which platform/arch/versions.modules values you have by launching a node.js instance and pasting the following statements:
`$ process.platform`
`$ process.arch`
`$ process.versions.modules`

**Comment by BCE8E8** [comment]
I typed these commands into the cmd of the node and it appeared:
process.platform = 'win32'
process.arch = 'x64'
process.versions.module = '57'

What I have to do? Do you have to install a version of these that appear in github? https://github.com/Robot/robot-js-binaries/tree/gh-pages/addon/1.0.2

Do I have to install the larger number? like, darwin-x64-48.node?

**Comment by karliky** [comment]
Your Node.js/npm should match any of the versions listed here https://github.com/Robot/robot-js-binaries/tree/gh-pages/addon/1.0.2 .

You may probable need to downgrade to Node.js 6 until we merge the dev branch into master which will fix this issue. Or you can take the code at the dev branch and execute the node-gyp installation command.

**Comment by BCE8E8** [comment]
Thank you very much, solved by downgrading.

**Comment by quinton-ashley** [comment]
Why has this still not been merged?  It's ridiculous to be telling people to downgrade from the LTS, how lazy.

**Comment by quinton-ashley** [comment]
Hey I found a solution to the problem:
[https://github.com/octalmage/robotjs](https://github.com/octalmage/robotjs)

**Comment by dkrutsko** [comment]
This issue will be fixed with the release of version 2.0.0.

Moreover, I think `octalmage/robotjs` is a very good package that should definitely be used if it solves your problem. `robot/robot-js` is targeted at a different audience that needs support for process manipulation rather than just keyboard and mouse automation.

As for the lack of support of this project, you're absolutely right, it is ridiculous to be telling people to downgrade and there is no excuse for my poor maintenance. With version 2.0.0 I have taken steps to improve the maintainability of this project and will try my best to keep up with the ever changing climate of node.js.

Thanks again for your feedback and I'm sorry if I let you down.
